### PR TITLE
Updated Implementation and Unit Testing for Kennard-Stone Sampler

### LINKED ISSUE #18: Add Tests for Kennard-Stone Sampler
The Kennard Stone sampling algorithm has already been implemented, but needs unit tests.

### PR DESCRIPTION
**Pull request by JacksonBurns**
Resolves #18

### CODE REVIEW

File: astartes/samplers/interpolation/kennardstone.py [hunk 1]
```diff
@@ -1,15 +1,16 @@
 from astartes.samplers import AbstractSampler
 
 # https://github.com/yu9824/kennard_stone
-from kennard_stone import train_test_split
+from kennard_stone import train_test_split as ks_train_test_split
+
+import numpy as np
```
Comment (kspieks) [inline astartes/samplers/interpolation/kennardstone.py:6]: I typically use a different order for imports e.g. placing standard libraries like numpy first, then related 3rd party imports like KS, then local imports from astartes. Just curious, what guidelines would you like to follow for this repo?

File: astartes/samplers/interpolation/kennardstone.py [hunk 2]
```diff
@@ -18,17 +19,9 @@ def _ks_split(self):
         set, so we ask for all but one of the points, and then put that
         index into the list at the end to circumvent this.
         """
-        _, _, samples_idxs, spare_idx = train_test_split(
+        _, _, samples_idxs, spare_idx = ks_train_test_split(
             self.X,
```
Comment (kspieks) [inline astartes/samplers/interpolation/kennardstone.py:23]: I believe we can remove self.X and only pass the list of indices (i.e. `list(range(len(self.X)))`) since splitting the X object is simply assigned to the throw away variables of `_`.

File: test/samplers/test_kennard_stone.py [hunk 3]
```diff
@@ -14,11 +16,109 @@ class Test_kennard_stone(unittest.TestCase):
...
+        )
+        self.assertIsNone(
+            np.testing.assert_array_equal(
+                labels_test,
+                np.array(["two"]),
+            ),
+            "Test labels incorrect.",
+        )
 
     def test_kennard_stone(self):
```
Comment (kspieks) [inline test/samplers/test_kennard_stone.py:98]: Sorry to be overly pedantic. Would it make sense to place this test first in the class so that it's run first? Seems like it's the most basic unit test to check before running our wrapper on it.

**Review by kspieks — CHANGES_REQUESTED** [review]
Thanks for adding these tests and reorganizing the KS Sampler! I think it looks pretty good but I had a few minor questions/comments. Happy to discuss in person today.

File: astartes/samplers/interpolation/kennardstone.py (hunk 2 above)
Comment (JacksonBurns) [reply astartes/samplers/interpolation/kennardstone.py:23]: The Kennard-Stone algorithm chooses the order of the samples based on the X array (sampling the extremes into the train set, and then interpolating for the test set) so self.X is required -- we ignore it because we ultimately just want to set _samples_idxs to indicate the order by which the samples should be withdrawn.

Throwing out the sampled X array is wasted work since we end up splitting it again, but the cost of doing so is low compared to the runtime of actually deciding on the sampling order.

File: astartes/samplers/interpolation/kennardstone.py (hunk 1 above)
Comment (JacksonBurns) [reply astartes/samplers/interpolation/kennardstone.py:6]: Agreed that we should settle on a configuration -- let's go with [`isort`](https://pycqa.github.io/isort/) which integrates nicely with the `black` formatter. I will open a separate PR to edit the action and update all the files.

File: test/samplers/test_kennard_stone.py (hunk 3 above)
Comment (JacksonBurns) [reply test/samplers/test_kennard_stone.py:98]: This test will actually be run first by default since unittest executes tests alphabetically, but I will move it first in the file to make that clearer. Agreed that it makes more sense.

File: astartes/samplers/interpolation/kennardstone.py (hunk 2 above)
Comment (kspieks) [reply astartes/samplers/interpolation/kennardstone.py:23]: My mistake. That makes sense. Thanks for explaining!

File: test/samplers/test_kennard_stone.py [hunk 4]
```diff
@@ -14,11 +16,109 @@ class Test_kennard_stone(unittest.TestCase):
...
+                y_train,
+                y_test,
+                labels_train,
+                labels_test,
+            ) = train_test_split(
+                self.X,
+                self.y,
+                labels=self.labels,
+                test_size=0.25,
+                train_size=0.75,
```
[the remark this answers is not in the document]
Comment (JacksonBurns) [reply test/samplers/test_kennard_stone.py:51]: I will do this and add a second test

**Review by JacksonBurns — COMMENTED** [review]
Unit tests have been re-ordered, separate PR opened for the import order issue.

**Review by kspieks — APPROVED** [review]
Thanks for incorporating the changes! This is ready to merge
